src/commands/new.rs

```diff
@@ -124,6 +134,77 @@ fn merge_values(from_file: &mut HashMap<String, String>, from_cli: &[ParameterVa
     }
 }
 
+async fn prompt_template(template_manager: &TemplateManager) -> anyhow::Result<Option<Template>> {
+    let mut templates = match get_or_install_templates(template_manager).await? {
+        Some(t) => t,
+        None => return Ok(None),
+    };
+    let opts = templates
+        .iter()
+        .map(|t| format!("{} ({})", t.id(), t.description_or_empty()))
+        .collect::<Vec<_>>();
+    let index = match dialoguer::Select::new()
+        .with_prompt("Pick a template to start your project with")
+        .items(&opts)
+        .default(0)
+        .interact_opt()?
+    {
+        Some(i) => i,
+        None => return Ok(None),
+    };
+    let choice = templates.swap_remove(index);
+    Ok(Some(choice))
+}
+
+const DEFAULT_TEMPLATES_INSTALL_PROMPT: &str =
+    "You don't have any templates yet. Would you like to install the default set?";
+const DEFAULT_TEMPLATE_REPO: &str = "https://github.com/fermyon/spin";
+
+async fn get_or_install_templates(
+    template_manager: &TemplateManager,
+) -> anyhow::Result<Option<Vec<Template>>> {
+    let templates = template_manager.list().await?.templates;
+    if templates.is_empty() {
+        let should_install = dialoguer::Confirm::new()
+            .with_prompt(DEFAULT_TEMPLATES_INSTALL_PROMPT)
+            .default(true)
+            .interact_opt()?;
+        if should_install == Some(true) {
+            install_default_templates().await?;
+            Ok(Some(template_manager.list().await?.templates))
+        } else {
+            println!(
+                "You can install the default templates later with 'spin install --git {}'",
+                DEFAULT_TEMPLATE_REPO
+            );
+            Ok(None)
+        }
+    } else {
+        Ok(Some(templates))
+    }
+}
+
+async fn install_default_templates() -> Result<(), anyhow::Error> {
+    let install_cmd = super::templates::Install {
+        git: Some(DEFAULT_TEMPLATE_REPO.to_owned()),
+        branch: None,
+        dir: None,
+        update: false,
+    };
+    install_cmd
+        .run()
+        .await
+        .context("Failed to install the default templates")?;
+    Ok(())
+}
+
+async fn prompt_name() -> anyhow::Result<String> {
+    let mut input = dialoguer::Input::<String>::new();
+    input.with_prompt("Enter a name for your new project");
+    let result = input.interact_text()?;
+    Ok(result)
```

Comment:
I wonder if this should loop on empty input, printing something like:

```
Project name is required. (ctrl-c to cancel)
```

Reply:
Right now, looks like the dialoguer doesnt continue until something is entered, which seems like proper behavior

Reply:
Looks like the `dialoguer` component already rejects empty input, though it does accept "all spaces."  (Which, to my surprise, we actually cope with.)  I'll trim and loop for that case just in case (though realistically we're more interested in protecting against Murphy vs. against Machiavelli).